admin/python/lsst/qserv/admin/nodePool.py

```diff
@@ -0,0 +1,160 @@
+# LSST Data Management System
+# Copyright 2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+Module defining nodePool class and related methods.
+
+NodePool is a class which allow to launch parallel ssh commands to
+a set of NodeAdmin instances. Each NodeAdmin instance can be a Qserv
+master or worker node.
+
+It has been used at CC-IN2P3 in spring 2015 to install and manage multiple
+Qserv development versions for a 50 nodes cluster.
+
+@author  Fabrice Jammes, IN2P3
+"""
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import logging
+import os
+from subprocess import Popen, list2cmdline
```

Comment:
Extract of /ust/lib/python2.7/subprocess.py:
def list2cmdline(seq):
    """
    Translate a sequence of arguments into a command line
    string, using the same rules as the MS C runtime:

    1) Arguments are delimited by white space, which is either a
       space or a tab.

    2) A string surrounded by double quotation marks is
       interpreted as a single argument, regardless of white space
       contained within.  A quoted string can be embedded in an
       argument.

    3) A double quotation mark preceded by a backslash is
       interpreted as a literal double quotation mark.

    4) Backslashes are interpreted literally, unless they
       immediately precede a double quotation mark.

    5) If backslashes immediately precede a double quotation mark,
       every pair of backslashes is interpreted as a literal
       backslash.  If the number of backslashes is odd, the last
       backslash escapes the next double quotation mark as
       described in rule 3.
    """